### DISCUSSION TITLE
Delayed Loading Spinner

### DISCUSSION THREAD

**Discussion by mvenghaus** [magewirephp/magewire#87] [Ideas]
Showing the loader is a good idea. But when the request is fast the loading spinner is more like a flicker. So my idea is to only show the loader when the requests takes more than 250ms.

**Comment by wpoortman** [comment]
I think it will create more flickering when you put a delay on it. Because most requests (on my machine) are around 250-400ms. So when you start it after 250, I think it will create even more flickering. It's the reason why I've added transitions to the loading spinner + overlay, to avoid flickering. Did you implement a custom loading spinner/overlay?

Curious how other look at this or what other options we would have. Maybe some system that remembers the average loading time for a request based on the component id and if its < 250, it should not show the loader #brainfart 😄

**Comment by mvenghaus** [comment]
indeed .. i hava a custom loader :)

maybe you are right .. the magewire component should decide whether the loader is shown or not .. usually you know which requests are fast and which may not
